import "common.proto";
import "hero.proto";

//请求武将列表 command:101
message GetHerosResponse
{
    repeated HeroPB heros = 1;
}

//武将升级 command:103
message HeroUpgradeResponse
{
    required CommonResponse res = 1;
    optional int32 level = 2;
    optional int32 exp = 3;
}

//武将突破 command:104
message HeroBreakResponse
{
    required CommonResponse res = 1;
    optional GameResourcesResponse consume = 2;
    optional int32 break_level = 3;
}

//武将献祭 command:105
message HeroSacrificeResponse
{
    required CommonResponse res = 1;
    optional GameResourcesResponse gain = 2;
}

//武将合成 command:106
message HeroComposeResponse
{
    required CommonResponse res = 1;
    optional HeroPB hero = 2;
}

//武将出售 command:107
message HeroSellResponse
{
    required CommonResponse res = 1;
    optional GameResourcesResponse gain = 2;
}

message HeroRefineResponse
{
    required CommonResponse res = 1;
    optional GameResourcesResponse consume = 2;
}

// 武将一键升级返回，command:119
message OneKeyHeroUpgradeRespone
{
    required CommonResponse res = 1;
    optional int32 level = 2;
    optional int32 exp = 3;
    repeated int32 exp_item_num = 4; // 消耗的经验丹 1大经验丹2中经验丹3小经验丹
}
